src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -32,206 +49,326 @@
                 gLogger.notice("To continue use token you need to add /DIRAC/Security/Authorization/issuer option.")
                 if not self.issuer:
                     DIRAC.exit(1)
                 DIRAC.gConfig.setOptionValue("/DIRAC/Security/Authorization/issuer", self.issuer)
 
-        # Try to get user information
+            # Try to get user authorization information from token
+            result = readTokenFromFile(self.outputFile)
+            if not result["OK"]:
+                return result
+            gLogger.notice(result["Value"].getInfoAsString())
+
+        return S_OK()
+
+    def loginWithCertificate(self):
+        """Login with certificate"""
+        # Search certificate and key
+        if not self.certLoc or not self.keyLoc:
+            cakLoc = Locations.getCertificateAndKeyLocation()
+            if not cakLoc:
+                return S_ERROR("Can't find user certificate and key")
+            self.certLoc = self.certLoc or cakLoc[0]
+            self.keyLoc = self.keyLoc or cakLoc[1]
+        # Generate proxy
+        self.outputFile = self.outputFile or Locations.getDefaultProxyLocation()
+        chain = X509Chain()
+        # Load user cert and key
+        result = chain.loadChainFromFile(self.certLoc)
+        if result["OK"]:
+            result = chain.loadKeyFromFile(self.keyLoc, password=getpass.getpass("Enter Certificate password:"))
+        if not result["OK"]:
+            return result
+
+        # Remember a clean proxy to then upload it in step 2
+        proxy = copy.copy(chain)
+
+        # Create local proxy with group
+        result = chain.generateProxyToFile(self.outputFile, int(self.lifetime) * 3600, self.group)
+        if not result["OK"]:
+            return S_ERROR(f"Couldn't generate proxy: {result['Message']}")
+
         result = Script.enableCS()
         if not result["OK"]:
             return S_ERROR("Cannot contact CS.")
-        DIRAC.gConfig.forceRefresh()
+        gConfig.forceRefresh()
 
-        if self.proxy:
-            result = getProxyInfo(self.proxyLoc)
-            if not result["OK"]:
-                return result["Message"]
-            gLogger.notice(formatProxyInfoAsString(result["Value"]))
+        # Step 2: Upload proxy to DIRAC server
+        return gProxyManager.uploadProxy(proxy)
+
+    def howToSwitch(self) -> bool:
+        """Helper message, how to switch access type(proxy or access token)"""
+        if "DIRAC_USE_ACCESS_TOKEN" in os.environ:
+            src, useTokens = ("env", os.environ.get("DIRAC_USE_ACCESS_TOKEN", "false").lower() in ("y", "yes", "true"))
         else:
-            result = readTokenFromFile(tokenFile)
-            if not result["OK"]:
-                return result
-            gLogger.notice(result["Value"].getInfoAsString())
+            src, useTokens = (
+                "conf",
+                gConfig.getValue("/DIRAC/Security/UseTokens", "false").lower() in ("y", "yes", "true"),
```

Comment:
This is a new option I think. Is it documented?

Reply:
In the code and in #5548 for example. And Ill add description to `DIRAC/docs/source/AdministratorGuide/Configuration/ConfReference/DiracSection.rst`.

Reply:
added